### PR TITLE
Quick Start for Existing Users V2-Changes to existing task-Preview Site 

### PR DESCRIPTION
**Pull request by AjeshRPai**
Parent #16349 

This PR has the following changes 
-  Updates title, subtitle of `View your Site` task 
-  Update Notice for `View your Site` task 
- Changes Focus Point to Site URL rather than `View Site` list item.
- Additional Updates Site Title Focus Point to the Start of the Site Title. (Before it was at the end of the Site title which didn't look proper for a short Site Title. So have updated the position to the top left corner of the Site Title)

| Site URL QS Focus point | Site Title Focus Point | 
|---|---|
| SiteURL_QS  | SiteTitleQS  | 

/cc @jostnes

To test:
- Launch the app with a site having a quick start in progress.
- Go to My Site tab.
- Select **Customize your site** from Next Steps card on My Site screen. The card will be present on the Initial Screen tab set in the Settings.
- Notice that title, subtitle of the `View your site` task are changed to `Visit your site`,`Preview your site to see what your visitors will see.`
- Click `View your site` row.
- Notice that snackbar text is changed to Select `Tap *Site Url* to view your site`.
- Notice that Focus point is shown on Site URL in the Site header 
- To test Notice with Go Button info, restart QS flow by clearing cache/ relogin. Run all QS tasks sequentially till you reach the **`Visit your site: Preview your site to see what your visitors will see`**. notice.

## Regression Notes
1. Potential unintended areas of impact
View your site title task not shown properly 

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual tests and Unit tests 

3. What automated tests I added (or what prevented me from doing so)
Unit tests 

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by AjeshRPai** [comment]
@osullivanchris Pinging you here so that I might get some inputs from you regarding the QS focus point position for **Check your Site title** and **View your Site** quick start task.  🤔 👁️

**Comment by osullivanchris** [comment]
@AjeshRPai  hey there! I installed the build and both these tasks look fine. Only idea is we could move the focus point 8dp or 16dp to the right. When its completely at the left edge, it starts looking close to other pieces of UI like the site icon.

**Comment by AjeshRPai** [comment]
@osullivanchris Yep, You are right. I moved the focus points 16dp to the right. Please have a look at it now.  

| Site Title  | URL  | 
|---|---|
|FocusPoint_SiteTitle| FocusPoint_SiteURl |

**Comment by osullivanchris** [comment]
> @osullivanchris Yep, You are right. I moved the focus points 16dp to the right. Please have a look at it now.

Thanks @AjeshRPai  looks good. What do you think? 

One other thought. The Site Title task is offset slightly above the text. Should we push the URL one downwards similarly, to show a bit more text and make clear you're tapping the bottom line?

**Comment by AjeshRPai** [comment]
>Should we push the URL one downwards similarly, to show a bit more text and make clear you're tapping the bottom line?

Hey @osullivanchris, at the moment the margin between the site title and URL feels just enough to show that they are related or together. Also when the Site title is long, then **2 lines of Title+ Margin + 1 line of the URL** is the max that we can show on the site header without making the vertical height more than the site icon. So My opinion would be not to move the URL downwards. 

One thing that I tried was to decrease the small size of the QS focus point and move it below the Site URL by a bit(4dp). It looks a little bit better now. Then again, if the Site title is long then the focus point would be on top of the Site URL. Let me know what you think.  🤔 . Does this look better?. 

| When the Site Title is short  | When the Site Title is long |  
|---|---|
|Screenshot_20220425_182754|Screenshot_20220425_182718|

**Comment by osullivanchris** [comment]
> One thing that I tried was to decrease the small size of the QS focus point and move it below the Site URL by a bit(4dp). It looks a little bit better now. Then again, if the Site title is long then the focus point would be on top of the Site URL. Let me know what you think. 🤔 . Does this look better?.

Hey I think we're getting into too much complexity now for any added value. Sounds like not worth changing. By the way just for clarity I didn't want to move the URL, it was the focus point on the URL.

But anyway lets just stick with the slight horizontal tweak that we already made and leave this change out! It was fine already.

**Comment by AjeshRPai** [comment]
>Hey I think we're getting into too much complexity now for any added value. Sounds like not worth changing. By the way just for clarity I didn't want to move the URL, it was the focus point on the URL.

@osullivanchris Yeah good point. 👍🏼 .To clarify - **I haven't made any changes to the position of the Site URL**. Just have the decreased the size of the focus point and moved it down by 4 dp.

File: WordPress/src/test/java/org/wordpress/android/ui/mysite/MySiteCardAndItemTest.kt [hunk 1]
```diff
@@ -32,6 +32,13 @@ class MySiteCardAndItemTest {
         assertThat(siteInfoCard.activeQuickStartItem).isTrue()
     }
 
+    @Test
+    fun `given site info block is active when focus point on subtitle`() {
```
Comment (ashiagr) [inline WordPress/src/test/java/org/wordpress/android/ui/mysite/MySiteCardAndItemTest.kt]: Nitpick: I think we can either remove `given` from the test description or convert it to a proper `given-when-then` format.

File: WordPress/src/test/java/org/wordpress/android/ui/mysite/MySiteViewModelTest.kt [hunk 2]
```diff
@@ -848,9 +848,10 @@ class MySiteViewModelTest : BaseUnitTest() {
     }
 
     @Test
-    fun `site info card url click opens site`() = test {
+    fun `site info card url click opens site and requests next step of View Site task`() = test {
         invokeSiteInfoCardAction(SiteInfoHeaderCardAction.URL_CLICK)
 
+        verify(quickStartRepository).completeTask(QuickStartTask.VIEW_SITE)
         assertThat(navigationActions).containsOnly(SiteNavigationAction.OpenSite(site))
```
Comment (ashiagr) [inline WordPress/src/test/java/org/wordpress/android/ui/mysite/MySiteViewModelTest.kt]: Nitpick: Considering that there's no next step for `View Site` task, how about we change `and requests next step of View Site task` to `and completes View Site task`?

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteViewModel.kt [hunk 3]
```diff
@@ -587,20 +584,28 @@ class MySiteViewModel @Inject constructor(
...
             _activeTaskPosition.postValue(quickStartTask to position)
         }
     }
 
-    private fun isSiteHeaderQuickStartTask(quickStartTask: QuickStartTask, position: Int): Boolean {
-        return if (position == LIST_INDEX_NO_ACTIVE_QUICK_START_ITEM &&
-                (quickStartTask == UPDATE_SITE_TITLE || quickStartTask == UPLOAD_SITE_ICON)) true
+    private fun isValidQuickStartFocusPosition(quickStartTask: QuickStartTask, position: Int): Boolean {
+        return if (position == LIST_INDEX_NO_ACTIVE_QUICK_START_ITEM && isSiteHeaderQuickStartTask(quickStartTask)) true
         else position >= 0
```
Comment (ashiagr) [inline WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteViewModel.kt]: Nitpick: I got this feedback on one of my PRs, and seeing its benefits in terms of readability sharing it with you:

> When you have an `if/else`, either both are on the same line without brackets or if there are in separate like then preferably add brackets on both.

**Review by ashiagr — APPROVED** [review]
Works as described! Just left non-blocking super-minor comments.  
Feel free to merge once they are addressed. 🙂

File: WordPress/src/test/java/org/wordpress/android/ui/mysite/MySiteCardAndItemTest.kt (hunk 1 above)
Comment (AjeshRPai) [reply WordPress/src/test/java/org/wordpress/android/ui/mysite/MySiteCardAndItemTest.kt]: @ashiagr Yup, I agree. Don't know how I missed that. Have changed the test description to a proper `given-when-then` in [430e39](https://github.com/wordpress-mobile/WordPress-Android/pull/16373/commits/d430e3947089f4c2d57d0ce7091a7202452e5cb5)

File: WordPress/src/test/java/org/wordpress/android/ui/mysite/MySiteViewModelTest.kt (hunk 2 above)
Comment (AjeshRPai) [reply WordPress/src/test/java/org/wordpress/android/ui/mysite/MySiteViewModelTest.kt]: @ashiagr Yup, you are right. Makes sense. 👍🏼 🙌🏼 . Have done the change in  [430e39](https://github.com/wordpress-mobile/WordPress-Android/pull/16373/commits/d430e3947089f4c2d57d0ce7091a7202452e5cb5) and [900fe8d](https://github.com/wordpress-mobile/WordPress-Android/pull/16373/commits/900fe8dd0f962de0ae90befc44b3e04ec62934ec)

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteViewModel.kt (hunk 3 above)
Comment (AjeshRPai) [reply WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteViewModel.kt]: @ashiagr Yup, agree. Will try to keep this in mind and follow it. I have done the changes in [c992906](https://github.com/wordpress-mobile/WordPress-Android/pull/16373/commits/c99290688e62e6c29d90eaa179f1939820de39ee). While formatting, I came across a similar `if-else`which I formatted. It had curly braces in the same line. Let me know if I should revert it.

Comment (ashiagr) [reply WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteViewModel.kt]: >  While formatting, I came across a similar if-elsewhich I formatted. It had curly braces in the same line. Let me know if I should revert it.

We could have just removed curly braces and kept this `if-else` in one line. 
> When you have an if/else, either both are on the same line without brackets ...

Comment (AjeshRPai) [reply WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteViewModel.kt]: 👍🏼  Done in [6178ee9](https://github.com/wordpress-mobile/WordPress-Android/pull/16373/commits/6178ee92d7ac970ac59016c5ef05b577083ae28d)

**Comment by AjeshRPai** [comment]
@ashiagr Thanks for the review and suggestions. 👍🏼 🙌🏼

**Pull request merged**